SourceCode/GPS/IO/TrackFiles.cs

```
@@ -0,0 +1,158 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.IO;
+using AgOpenGPS.Core.Models;
+
+namespace AgOpenGPS.IO
+{
+    public static class TrackFiles
+    {
+        public static List<CTrk> Load(string fieldDirectory)
+        {
+            var result = new List<CTrk>();
+            var path = Path.Combine(fieldDirectory, "TrackLines.txt");
+            if (!File.Exists(path)) return result;
+
+            using (var reader = new StreamReader(path))
+            {
+                // Skip optional header
+                var firstLine = reader.ReadLine();
+                if (firstLine == null) return result;
+                if (!firstLine.TrimStart().StartsWith("$", StringComparison.Ordinal))
+                {
+                    // rewind logic: treat this line as first "name"
+                    reader.BaseStream.Seek(0, SeekOrigin.Begin);
+                    reader.DiscardBufferedData();
+                }
+
+                while (!reader.EndOfStream)
+                {
+                    // --- Name ---
+                    var name = (reader.ReadLine() ?? string.Empty).Trim();
+                    if (string.IsNullOrEmpty(name)) continue;
+
+                    // --- Average heading ---
+                    if (reader.EndOfStream) break;
+                    var headingLine = reader.ReadLine() ?? string.Empty;
+                    double avgHeading;
+                    if (!double.TryParse(headingLine.Trim(), NumberStyles.Float, CultureInfo.InvariantCulture, out avgHeading))
```

I'm actually not sure if it is a good idea to use `TryParse` everywhere instead of `Parse`. That way we silently ignore invalid values instead of complaining about it... is that really what we want?